app/src/main/java/org/mozilla/focus/fragment/DownloadDialogFragment.kt

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.fragment
+
+import android.os.Build
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import android.support.v7.app.AlertDialog
+import android.text.Html
+import android.text.Spanned
+import android.view.View
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import mozilla.components.support.utils.DownloadUtils
+import org.mozilla.focus.R
+import org.mozilla.focus.telemetry.TelemetryWrapper
+import org.mozilla.focus.web.Download
+
+/**
+ * Fragment displaying a download dialog
+ */
+class DownloadDialogFragment : DialogFragment() {
+
+    override fun onCreateDialog(bundle: Bundle?): AlertDialog {
```

Comment:
\[detekt\] INFO: \[complexity/LongMethod\] One method should have one responsibility\. Long methods tend to handle many things at once\. Prefer smaller methods to make them easier to understand\.